pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -43,10 +45,19 @@ func NewTerminator(clk clock.Clock, kubeClient client.Client, eq *Queue) *Termin
 	}
 }
 
-// Cordon cordons a node
+// Cordon adds the karpenter.sh/disruption taint to a node with a NodeClaim
+// If the node is linked to a machine, it will use the node.kubernetes.io/unschedulable taint.
 func (t *Terminator) Cordon(ctx context.Context, node *v1.Node) error {
 	stored := node.DeepCopy()
-	node.Spec.Unschedulable = true
+
+	if _, isMachine := node.Labels[v1alpha5.ProvisionerNameLabelKey]; isMachine {
+		node.Spec.Unschedulable = true
+	} else {
+		node.Spec.Taints = lo.Reject(node.Spec.Taints, func(t v1.Taint, _ int) bool {
+			return t.Key == v1beta1.DisruptionTaintKey
```

Comment:
Should we leverage our `v1beta1.IsDisruptingTaint` function that you created here?

Reply:
Yeah I had just done the key here since it should apply to the future taints as well, but since we haven't added that in, I'll revert to the function call 👍